File: ruby/lib/enolib/elements/missing/missing_field.rb
# frozen_string_literal: true

module Enolib
  class MissingField < MissingValueElementBase
    def attribute(key = nil)
      MissingAttribute.new(key, self)
    end
    
    def attributes(_key = nil)
      []
    end
    
    def items
      []
    end
    
    def optional_attribute(_key = nil)
      nil
    end
    
    def optional_string_values(_key = nil)
      []
    end
    
    def optional_values(_loader = nil)
      []
    end
    
    def required_attribute(_key = nil)
      @parent._missing_error(self)
    end
    
    def required_string_values(_key = nil)
      []
    end
    
    def required_values(_loader = nil)
      []
    end
    
    def to_s
      if @key
        "#<Enolib::MissingField key=#{@key}>"
      else
        '#<Enolib::MissingField>'
      end
    end
  end
end

File: python/enolib/elements/missing/missing_field.py
from .missing_value_element_base import MissingValueElementBase
from .missing_attribute import MissingAttribute

class MissingField(MissingValueElementBase):
    def __repr__(self):
        if self._key:
            return f"<class MissingField key={self._key}>"

        return '<class MissingField>'

    def attribute(self, key=None):
        return MissingAttribute(key, self)
        
    def attributes(self, _key=None):
        return []

    def items(self):
        return []

    def optional_attribute(self, _key=None):
        return None
        
    def optional_string_values(self):
        return []

    def optional_values(self, _loader):
        return []

    def required_attribute(self, _key=None):
        self._parent._missing_error(self)

    def required_string_values(self):
        return []

    def required_values(self, _loader):
        return []
